client/components/Dashboard.jsx

```diff
@@ -3,32 +3,109 @@ import { useNavigate } from "react-router-dom";
 
 export default function Dashboard() {
   const [user, setUser] = useState(null);
+  const [today, setToday] = useState(null);
   const nav = useNavigate();
 
   useEffect(() => {
     fetch(`${import.meta.env.VITE_API_URL}/auth/me`, { credentials: "include" })
       .then(res  => res.json())
       .then(data => setUser(data))
-      .catch(() => {
-        nav("/login");
-      });
+      .catch(() => {});
+
+    const now = new Date();
+    const formatted = now.toLocaleDateString("en-US", {
+      weekday: "long", month: "long", day: "numeric",});
+    setToday(formatted);
   }, []);
 
   if (!user) return <p>Loading your profile…</p>;
 
   const logout = () => {
     fetch(`${import.meta.env.VITE_API_URL}/auth/logout`, {
-      method:      "POST",
+      method: "POST",
       credentials: "include",
     }).then(() => {
       nav("/login");
     });
   };
 
   return (
-    <div>
-      <h1>Welcome, {user.firstName || user.email}!</h1>
-      <button onClick={logout}>Logout</button>
+    <div className="dashboard-container">
+      <header className="dashboard-banner">
+        <h1>Welcome, {user.firstName}</h1>
+      </header>
+
+      <div className="dashboard-meta">
+        <span className="dashboard-date">{today}</span>
+        <span className="dashboard-streak">
+          🔥 Streak: {user.dailyStreak || 0} days
+        </span>
+      </div>
+
+      <div className="dashboard-actions">
+        <button className="btn primary">Log Meal</button>
+        <button className="btn primary">Log Workout</button>
+      </div>
+
+      <div className="dashboard-summary">
+        <div className="card">
+          <h2>Calories In</h2>
+          <p>{user.dailyCaloriesIn || 0} kcal</p>
+        </div>
+        <div className="card">
+          <h2>Calories Burned</h2>
+          <p>{user.dailyCaloriesOut || 0} kcal</p>
+        </div>
+        <div className="card">
+          <h2>Meals Logged</h2>
+          <p>{user.mealsCount || 0}</p>
+        </div>
+        <div className="card">
+          <h2>Workouts Logged</h2>
+          <p>{user.workoutsCount || 0}</p>
+        </div>
+      </div>
+
+      <div className="dashboard-progress">
+        <h2>Daily Calorie Goal</h2>
+        <div className="progress-bar">
+          <div
+            className="progress"
+            style={{
+              width: user.calorieGoal
+                ? `${(user.dailyCaloriesIn / user.calorieGoal) * 100}%`
+                : "0%",
```

Comment:
```suggestion
              width: typeof user.calorieGoal === 'number' && user.calorieGoal > 0
                ? `${(user.dailyCaloriesIn / user.calorieGoal) * 100}%`
                : "0%",
```
Are we already protecting against goal being 0?